src/components/BodyPortal.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { createPortal } from 'react-dom';
+import { BodyPortalSlotsContext } from './BodyPortalSlotsContext';
+
+const getInsertBeforeTarget = (bodyPortalSlots: string[], slot: string) => {
+  // Note: If the slot is not found in bodyPortalSlots, this code will insert the tag before the first slot
+  //       This is not recommended usage, as the ordering will then depend on the rendering order and may change
+  for (let index = bodyPortalSlots.findIndex((sl) => sl === slot) + 1; index < bodyPortalSlots.length; index++) {
+    const sl = bodyPortalSlots[index];
+    const tag = sl === 'main'
+    ? document.body.querySelector('main')
```

Comment:
Another option here is to find main by `id="root"` which mirrors what React does more closely, though I find `main` works better as the slot name... I'm not sure which is best.